app/models/mixins/compliance_mixin.rb

```diff
@@ -35,8 +37,15 @@ def scan_and_check_compliance_queue
   end
 
   def compliance_policies
-    target_class = self.class.base_model.name.downcase
-    _, plist = MiqPolicy.get_policies_for_target(self, "compliance", "#{target_class}_compliance_check")
+    _, plist = MiqPolicy.get_policies_for_target(self, "compliance", "#{compliance_policy_name}_compliance_check")
     plist
   end
+
+  def compliance_policy_name
```

Comment:
Originally I was thinking this would include the full name (i.e. `"#{compliance_policy_name}_compliance_check"`), which is why I suggested the method name `compliance_policy_name`.  However, I see that you're using it as more of a prefix since it is able to be reused elsewhere (which I like better), so I think this method should be instead be named `compliance_policy_prefix`

Reply:
agreed. I had this as a full name and then I found all the places that did the name derivation and since we have other names that are based upon this, I took the tangent.

thanks. fixed